internal/pkg/cli/svc_override.go

```
@@ -243,6 +255,22 @@ func (o *overrideSvcOpts) newSvcPackageCmd(tplBuf stringWriteCloser) (executor,
 	return cmd, nil
 }
 
+// targetEnvName returns the name of the environment to use when running "svc package".
+// If the user does not explicitly provide an environment, default to a random environment.
+func (o *overrideSvcOpts) targetEnvName() (string, error) {
+	if o.envName != "" {
+		return o.envName, nil
+	}
+	envs, err := o.cfgStore.ListEnvironments(o.appName)
+	if err != nil {
+		return "", fmt.Errorf("list environments in application %q: %v", o.appName, err)
+	}
+	if len(envs) == 0 {
+		return "", fmt.Errorf("no environments found in application %q", o.appName)
+	}
+	return envs[0].Name, nil
```

is this "random"?🤔 is the order from `ListEnvironments` deterministic?

---

You're right it's not "random", I just didn't want customers to rely on a particular environment for it. We should be able to change which env we default to if we want to.

It's always the first environment that they've created for now. Would you like me to update the flag description?

---

i was thinking that we don't want people to rely on the undocumented, deterministic behavior hiding potential errors - but with the context that this happens in `svc override`, which is just a best-effort scaffolding that's likely only run a single time, i think the behavior/description is perfectly fine 👌